### PR TITLE
Move ssh code to new package

### PR DESCRIPTION
**Pull request by johnSchnake**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
The framework/ssh.go code was heavily used throughout the framework
and could be useful elsewhere but reusing those methods requires
importing all of the framework.

Extracting these methods to their own package for reuse.

Only a few methods had to be copied into this package from the
rest of the framework to avoid an import cycle.

**Which issue(s) this PR fixes**:
Ref #77095 

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

### CODE REVIEW

**Comment by johnSchnake** [comment]
Currently just WIP, hence my own hold. Just moved everything to a new pkg and now need to try and update refs.

**Comment by alejandrox1** [comment]
Minor comment: you reference #77905 in the PR message. I don't think that issue exists :sweat_smile:

**Comment by johnSchnake** [comment]
Thanks; I transposed 2 digits. Sorry about that.

**Review by timothysc — COMMENTED** [review]
/assign @timothysc

File: test/e2e/apimachinery/BUILD
Comment (johnSchnake) [inline test/e2e/apimachinery/BUILD]: pulled in github dep; will fix.

File: test/e2e/framework/profile_gatherer.go
Comment (fejta) [inline test/e2e/framework/profile_gatherer.go]: It looks like every import aliases the package name. Did we choose the wrong name?

File: test/e2e/framework/ssh/ssh.go
Comment (fejta) [inline test/e2e/framework/ssh/ssh.go]: I am not a fan of aliasing everything. There should be good and exceptional reasons to override the package name.

File: test/e2e/storage/flexvolume.go
Comment (fejta) [inline test/e2e/storage/flexvolume.go]: corev1

Comment (johnSchnake) [reply test/e2e/storage/flexvolume.go]: This is just a goimports change; not an attempt to really alias it. Do you really want me to change the default alias here or you were just disagreeing with my the idea that I explicitly chose one (and it was v1)?

By the way, this is the motivation for https://github.com/kubernetes/kubernetes/pull/77407 ; the change has nothing to do with this PR but is inevitably discussed.

File: test/e2e/framework/profile_gatherer.go
Comment (johnSchnake) [reply test/e2e/framework/profile_gatherer.go]: It was an explicit choice of the testing-commons group; both the name of the package and the alias; I actually put up the PR up without and was told to alias everything.

It is compatible with all the logging PRs that are going in where the package name was `log` but we are explicitly aliasing everything with `e2elog`.

**Comment by timothysc** [comment]
@johnSchnake - needs rebase.

**Comment by johnSchnake** [comment]
@timothysc Yes it needs rebased but since the other PRs I had for refactoring were all in the same 'nearly merged' status I was going to wait to rebase until each of those went in to avoid rebrasing all N of my PRs over and over.

https://github.com/kubernetes/kubernetes/pull/77413 is about to merge (already approved)
https://github.com/kubernetes/kubernetes/pull/77408 just needs approval
Then I'll rebase this and need approval/lgtm again

File: test/e2e/storage/flexvolume.go
Comment (fejta) [reply test/e2e/storage/flexvolume.go]: Standard style is to call this corev1

https://github.com/kubernetes/kubernetes/search?utf8=%E2%9C%93&q=corev1&type=

File: test/e2e/framework/profile_gatherer.go
Comment (fejta) [reply test/e2e/framework/profile_gatherer.go]: Not a fan but this isn't a hill I need to die on.

File: test/e2e/storage/flexvolume.go
Comment (johnSchnake) [reply test/e2e/storage/flexvolume.go]: FWIW it looks like that is not quite the case. I'm happy to change it but this was why #77045 got created; `v1` and `corev1` get used roughly equally, but unaliased is about 5x as popular.

My approach on #77407 was to run goimports on a bunch of files to make them consistent but that would have pushed the `v1` alias. If you prefer the corev1 I can do that instead. (It is the goimports that adds the v1 alias, not the editor itself though I had heard that as well but confirmed goimports itself is the perpetrator)

```
$ rg '^[\s]*v1 "k8s.io/api/core/v1"' | wc -l
     220
$ rg '^[\s]*corev1 "k8s.io/api/core/v1"' | wc -l
     257
$ rg '^[\s]*"k8s.io/api/core/v1"' | wc -l
    1130
```

Comment (johnSchnake) [reply test/e2e/storage/flexvolume.go]: @fejta Are you OK if I leave the `v1` alias in _this_ PR and let #77045 handle squaring all those?

The problem is that I am touching 32 different files and if I want to add an alias to these it is going to probably double the size of this PR even though it has nothing to do with ssh.

The explicit `v1` alias didnt require code changes at each of the lines using something from that package but `corev1` is a much bigger change.

Comment (fejta) [reply test/e2e/storage/flexvolume.go]: Sure SGTM

File: test/e2e/framework/ssh/ssh.go
Comment (timothysc) [inline test/e2e/framework/ssh/ssh.go]: I'll let this pass, but I'd bet we have a bunch of overlap with other functions.

**Review by timothysc — APPROVED** [review]
/lgtm 
/approve

File: test/e2e/framework/ssh/ssh.go
Comment (johnSchnake) [reply test/e2e/framework/ssh/ssh.go]: It is 100% copied so I'm positive we do. It was part of the `framework` though where we are actively trying to remove code from. My understanding was that we expected a number of small methods like this to be duplicated across the codebase in order to support the decoupling.

**Pull request merged**
